src/rdkafka_request.c

```diff
@@ -5878,6 +5878,95 @@ rd_kafka_DeleteAclsRequest(rd_kafka_broker_t *rkb,
         return RD_KAFKA_RESP_ERR_NO_ERROR;
 }
 
+/**
+ * @brief Construct and send ElectLeadersRequest to \p rkb
+ *        with the partitions (ElectLeaders_t*) in \p elect_leaders, using
+ *        \p options.
+ *
+ *        The response (unparsed) will be enqueued on \p replyq
+ *        for handling by \p resp_cb (with \p opaque passed).
+ *
+ * @returns RD_KAFKA_RESP_ERR_NO_ERROR if the request was enqueued for
+ *          transmission, otherwise an error code and errstr will be
+ *          updated with a human readable error string.
+ */
+rd_kafka_resp_err_t rd_kafka_ElectLeadersRequest(
+    rd_kafka_broker_t *rkb,
+    const rd_list_t *elect_leaders /*(rd_kafka_EleactLeaders_t*)*/,
+    rd_kafka_AdminOptions_t *options,
+    char *errstr,
+    size_t errstr_size,
+    rd_kafka_replyq_t replyq,
+    rd_kafka_resp_cb_t *resp_cb,
+    void *opaque) {
+        rd_kafka_buf_t *rkbuf;
+        int16_t ApiVersion;
+        const rd_kafka_ElectLeaders_t *elect_leaders_request;
+        int op_timeout;
+
+        if (rd_list_cnt(elect_leaders) == 0) {
+                rd_snprintf(errstr, errstr_size,
+                            "No partitions specified for leader election");
+                rd_kafka_replyq_destroy(&replyq);
+                return RD_KAFKA_RESP_ERR__INVALID_ARG;
+        }
+
+        elect_leaders_request = rd_list_elem(elect_leaders, 0);
+
+        ApiVersion = rd_kafka_broker_ApiVersion_supported(
+            rkb, RD_KAFKAP_ElectLeaders, 0, 2, NULL);
+        if (ApiVersion == -1) {
+                rd_snprintf(errstr, errstr_size,
+                            "ElectLeaders Admin API (KIP-460) not supported "
+                            "by broker, requires broker version >= 2.4.0");
+                rd_kafka_replyq_destroy(&replyq);
+                return RD_KAFKA_RESP_ERR__UNSUPPORTED_FEATURE;
+        }
+
+        if (elect_leaders_request->partitions != NULL) {
+                rkbuf = rd_kafka_buf_new_flexver_request(
+                    rkb, RD_KAFKAP_ElectLeaders, 1,
+                    1 + (50 + 4) * elect_leaders_request->partitions->cnt,
+                    ApiVersion >= 2);
+        } else {
+                rkbuf = rd_kafka_buf_new_flexver_request(
+                    rkb, RD_KAFKAP_ElectLeaders, 1, 1 + 4, ApiVersion >= 2);
+        }
```

Comment:
Buf size will grow automatically, so this is an estimate, but you can use an app variable an have a single call to `rd_kafka_buf_new_flexver_request`.
```suggestion

        rd_buf_size_estimate =
            1 /* ElectionType */ + 4 /* #TopicPartitions */ + 4 /* TimeoutMs */;
        if (elect_leaders_request->partitions)
                rd_buf_size_estimate +=
                    (50 + 4) * elect_leaders_request->partitions->cnt;

        rkbuf = rd_kafka_buf_new_flexver_request(rkb, RD_KAFKAP_ElectLeaders, 1,
                                                 rd_buf_size_estimate,
                                                 ApiVersion >= 2);
```